patches/0006-LPS-118624-Don-t-pass-languageId-to-css-files-reques.patch

```diff
@@ -0,0 +1,25 @@
+From a62acb27ffd532f3c3bb00e9ad727b62a95e9af6 Mon Sep 17 00:00:00 2001
+From: Carlos Lancha <carlos.lancha@liferay.com>
+Date: Thu, 6 Aug 2020 14:42:21 +0200
+Subject: [PATCH 6/6] LPS-118624 Don't pass languageId to css files requests
+
+---
+ core/ckeditor_base.js | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/core/ckeditor_base.js b/core/ckeditor_base.js
+index a6ce37594..6d9fcbc03 100644
+--- a/core/ckeditor_base.js
++++ b/core/ckeditor_base.js
+@@ -184,7 +184,7 @@ if ( !window.CKEDITOR ) {
+ 						var key = entries[0];
+ 						var value = entries[1];
+ 
+-						if (!resourceUrl.searchParams.has(key)) {
```

Comment:
Patches are quite hard to review... but I'd say it'd be better abort before even starting the the loop sequence

```
Object.entries(CKEDITOR.ADDITIONAL_RESOURCE_PARAMS).forEach(function(entries) {
    ...
}
```

No reason to check the same for each key.

Reply:
> Patches are quite hard to review

Pro-Tip™: click the "View file" link (in the ... menu for the patch) to see the change without all the noise: https://github.com/liferay/liferay-ckeditor/blob/80e862ef5f0704180006404d229151f047e66688/patches/0006-LPS-118624-Don-t-pass-languageId-to-css-files-reques.patch

Reply:
Yeah, what I meant is you don't have the interactive diff, so if you want to see where is that `if` you have to check the source file, right? 

And you don't know what other patches are applied on top, so you need to check the source file plus there result of the previous patches to know for sure how does the surrounding code look like?

Reply:
M... but we want to not add the param only if the key is `languageId`. If we abort it before iterating we won't add any param

Reply:
Yeah, that's a fair point, I was thinking just skip all params for css files, but this should do the trick too

Reply:
It's a bit weirder because it's harder to explain that `if` now... people coming in the future (us) won't understand why simply by the message `fix: Don't pass languageId to css files requests`. We will understand what, but not what was so special about css files and the `languageId` param.

Reply:
Roger that

Reply:
> Yeah, what I meant is you don't have the interactive diff, so if you want to see where is that `if` you have to check the source file, right?
> 
> And you don't know what other patches are applied on top, so you need to check the source file plus there result of the previous patches to know for sure how does the surrounding code look like?

If it's a non-trivial patch, your best bet is probably to pull it down the PR locally and run `./ck.sh setup`. In the ckeditor-dev submodule you can then see the commits that the patches correspond to and explore the file-system etc. If it is non-trivial too, you may want to build and test anyway, so pulling it down might be something you want to do anyway.

There's a definite trade-off here. We explored a bunch of options when we switched to the patch-based workflow, which you can read about [here](https://github.com/liferay/liferay-ckeditor/issues/16). We can obviously try one of the other options on that list (or perhaps something not even on it), if this one is too painful.

Reply:
nah, it's okay, I was just passing by

Reply:
If you want to know real pain, read about [how it used to be](https://github.com/liferay/liferay-ckeditor/issues/7).

![pain](https://user-images.githubusercontent.com/7074/89542745-5ec83480-d800-11ea-9c40-2e063f09b584.gif)

Reply:
![lifeispain](https://user-images.githubusercontent.com/905006/89543293-05acd080-d801-11ea-940d-43573fbad524.gif)